crates/ntx-builder/src/actor/mod.rs

```diff
@@ -0,0 +1,339 @@
+
+/// The mode of operation that the account actor is currently performing.
+#[derive(Default, Debug)]
+enum ActorMode {
+    #[default]
+    NoViableNotes,
+    NotesAvailable,
+    TransactionInflight(TransactionId),
+}
+
+// ACCOUNT ACTOR
+// ================================================================================================
+
+/// A long-running asynchronous task that handles the complete lifecycle of network transaction
+/// processing. Each actor operates independently and is managed by a single coordinator that
+/// spawns, monitors, and messages all actors.
+///
+/// ## Core Responsibilities
+///
+/// - **State Management**: Loads and maintains the current state of network accounts, including
+///   available notes, pending transactions, and account commitments.
+/// - **Transaction Selection**: Selects viable notes and constructs a [`TransactionCandidate`]
+///   based on current chain state.
+/// - **Transaction Execution**: Executes selected transactions using either local or remote
+///   proving.
+/// - **Mempool Integration**: Listens for mempool events to stay synchronized with the network
+///   state and adjust behavior based on transaction confirmations.
+///
+/// ## Lifecycle
+///
+/// 1. **Initialization**: Loads account state from the store or uses provided account data.
+/// 2. **Event Loop**: Continuously processes mempool events and executes transactions.
+/// 3. **Transaction Processing**: Selects, executes, and proves transactions, and submits them to
+///    block producer.
+/// 4. **State Updates**: Updates internal state based on mempool events and execution results.
+/// 5. **Shutdown**: Terminates gracefully when cancelled or encounters unrecoverable errors.
+///
+/// ## Concurrency
+///
+/// Each actor runs in its own async task and communicates with other system components through
+/// channels and shared state. The actor uses a cancellation token for graceful shutdown
+/// coordination.
+pub struct AccountActor {
+    origin: AccountOrigin,
+    store: StoreClient,
+    mode: ActorMode,
+    event_rx: mpsc::Receiver<MempoolEvent>,
+    cancel_token: CancellationToken,
+    block_producer: BlockProducerClient,
+    prover: Option<RemoteTransactionProver>,
+    chain_state: Arc<RwLock<ChainState>>,
+}
+
+impl AccountActor {
+    /// Constructs a new account actor and corresponding messaging channel with the given
+    /// configuration.
+    pub fn new(
+        origin: AccountOrigin,
+        config: &AccountActorConfig,
+        event_rx: mpsc::Receiver<MempoolEvent>,
+        cancel_token: CancellationToken,
+    ) -> Self {
+        let block_producer = BlockProducerClient::new(config.block_producer_url.clone());
+        let prover = config.tx_prover_url.clone().map(RemoteTransactionProver::new);
+        Self {
+            origin,
+            store: config.store.clone(),
+            mode: ActorMode::default(),
```

Comment:
nit: I'd probably just use `ActorMode::NoViableNotes` and remove `Default` derive from `ActorMode`. This would make it a bit simpler to follow the code.